File: src/bmsx/audio/runtime_asset_audio.ts
import type { SoundMaster } from './soundmaster';
import type { AssetEntry } from '../machine/memory/memory';

export function syncDirtyRuntimeAudioAssets(dirtyAssets: Iterable<AssetEntry>, sndmaster: SoundMaster): void {
	for (const entry of dirtyAssets) {
		if (entry.type === 'audio') {
			sndmaster.invalidateClip(entry.id);
		}
	}
}

File: src/bmsx_cpp/audio/runtime_asset_audio.cpp
#include "audio/runtime_asset_audio.h"

#include "audio/soundmaster.h"
#include "machine/memory/memory.h"

namespace bmsx {
namespace RuntimeAssetAudio {

void syncDirtyRuntimeAudioAssets(const std::vector<Memory::AssetEntry*>& dirtyAssets, SoundMaster& soundMaster) {
	for (const auto* entry : dirtyAssets) {
		if (entry->type == Memory::AssetType::Audio) {
			soundMaster.invalidateClip(entry->id);
		}
	}
}

} // namespace RuntimeAssetAudio
} // namespace bmsx
